MigFiles/SupportLibraries/ZWaveLib/Controller.cs

```diff
@@ -337,9 +341,17 @@ private void ZwaveMessageReceived(object sender, ZWaveMessageReceivedEventArgs a
                                 var newNode = CreateDevice(nodeOperationIdCheck, 0x00);
                                 // Extract node information frame
                                 int nodeInfoLength = (int)args.Message[7];
-                                byte[] nodeInfo = new byte[nodeInfoLength - 2];
-                                Array.Copy(args.Message, 8, nodeInfo, 0, nodeInfoLength - 2);
+                                // we don't need to exclude the last 2 CommandClasses
+                                byte[] nodeInfo = new byte[nodeInfoLength];
+                                Array.Copy(args.Message, 8, nodeInfo, 0, nodeInfoLength);
                                 newNode.NodeInformationFrame = nodeInfo;
+
+                                newNode.BasicClass = args.Message[8];
+                                newNode.GenericClass = args.Message[9];
+                                newNode.SpecificClass = args.Message[10];
+                                devices.Add(newNode);
+                                newNode.HandleSecureCommandClasses(nodeInfo);
+
```

Comment:
when a code is included we need to do a handshake and set the new security key - That has to happen at the inclusion time.